packages/renderer/src/lib/kube/resource-listen.ts

```diff
@@ -0,0 +1,132 @@
+  // called every time the resource is updated
+  onResourceUpdated: (resource: KubernetesObject, experimental: boolean) => void;
+  // called every time events attached to the resource are updated
+  onEventsUpdated?: (events: CoreV1Event[]) => void;
+}
+
+/** listenResource listens for a specific resource (resourceName, name, namespace)
+ *  and fires events when:
+ *  - the resource is not found
+ *  - the resource is updated (will send the resource without managedFields)
+ *  - the events attached to the resource are updated, optionally
+ */
+export async function listenResource(options: ListenResourceOptions): Promise<IDisposable | undefined> {
+  let resourcesListener: IDisposable | undefined;
+  let eventsListener: IDisposable | undefined;
+
+  let resourcesUnsubscriber: Unsubscriber;
+  let eventsUnsubscriber: Unsubscriber;
+
+  let allEvents: KubernetesObject[] = [];
+  let resource: KubernetesObject | undefined;
+
+  const refreshEvents = (): void => {
+    if (options.listenEvents) {
+      const events = allEvents
+        .map(ev => (isCoreV1Event(ev) ? ev : undefined))
+        .filter(ev => !!ev)
+        .filter(ev => ev.involvedObject.uid === resource?.metadata?.uid);
+      options.onEventsUpdated?.(events);
+    }
+  };
+
+  const onUpdatedResources = (updatedResources: KubernetesObject[]): void => {
+    const matchingResource = updatedResources.find(
+      srv =>
+        srv.metadata?.name === options.name && (!options.namespace || srv.metadata?.namespace === options.namespace),
+    );
+    if (matchingResource) {
+      resource = matchingResource;
+      const kubeResource = nonVerbose(matchingResource);
+      options.onResourceUpdated(kubeResource, isExperimental);
+      refreshEvents();
+    } else {
+      options.onResourceNotFound();
+    }
+  };
+
+  const isExperimental = await isKubernetesExperimentalMode();
+  if (isExperimental) {
+    resourcesListener = await listenResources(options.resourceName, {}, onUpdatedResources);
+    if (options.listenEvents) {
+      eventsListener = await listenResources('events', {}, (updatedEvents: KubernetesObject[]) => {
+        allEvents = updatedEvents;
+        refreshEvents();
+      });
+    }
+  } else {
+    if (options.legacyResourceStore) {
+      resourcesUnsubscriber = options.legacyResourceStore.subscribe(onUpdatedResources);
+    }
+    if (options.listenEvents && options.legacyEventsStore) {
+      eventsUnsubscriber = options.legacyEventsStore.subscribe(updatedEvents => {
+        allEvents = updatedEvents.map(e => ({ kind: 'Event', ...e }));
+        refreshEvents();
+      });
+    }
+  }
+
+  return {
+    dispose: (): void => {
+      resourcesListener?.dispose();
+      eventsListener?.dispose();
+      resourcesUnsubscriber?.();
+      eventsUnsubscriber?.();
+    },
+  };
+}
+
+// removes "verbose" fields from a Kubernetes resource description:
+// - metadata.managedFields
+function nonVerbose(resource: KubernetesObject): KubernetesObject {
```

Comment:
question: why do we delete the managed fields ?

remark: maybe this filtering is something that should have been done on the `main` side ? To avoid the unnecessary payload size with the data to finally delete it.

Reply:
> question: why do we delete the managed fields ?

To have the same behaviour as currently (#7593). In non-experimental mode, it is removed when calling `loadDetails` (in main, see 2nd point)

> remark: maybe this filtering is something that should have been done on the main side ? To avoid the unnecessary payload size with the data to finally delete it.

In non-experimental, it is removed only when calling `loadDetails`, but is still present in other places. This information can be interesting for some developers, and could be displayed in a readable form to help us (debug operators for example). I'm trying to remove it only at this level, to keep it in other places.

Reply:
I see